l1-contracts/src/core/Rollup.sol

```diff
@@ -262,13 +271,51 @@ contract Rollup is EIP712("Aztec Rollup", "1"), Leonidas, IRollup, ITestRollup {
 
     // @note  Only if the rollup is the canonical will it be able to meaningfully claim fees
     //        Otherwise, the fees are unbacked #7938.
-    if (address(this) == FEE_JUICE_PORTAL.canonicalRollup()) {
-      for (uint256 i = 0; i < Constants.AZTEC_EPOCH_DURATION; i++) {
+    bool isFeeCanonical = address(this) == FEE_JUICE_PORTAL.canonicalRollup();
+    bool isSysstiaCanonical = address(this) == SYSSTIA.canonicalRollup();
```

Comment:
Would we ever want these to differ?

Reply:
Likely not. But done this way as they could. For example, it you want to run a Sysstia that allow rewards (of different size) to multiple rollups it can be supported this way, but there can only be one guy that is supported for the fee collateral.